File: src/main/java/org/apache/fineract/client/services/AccountNumberFormatApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.DeleteAccountNumberFormatsResponse;
import org.apache.fineract.client.models.GetAccountNumberFormatsIdResponse;
import org.apache.fineract.client.models.GetAccountNumberFormatsResponseTemplate;
import org.apache.fineract.client.models.PostAccountNumberFormatsRequest;
import org.apache.fineract.client.models.PostAccountNumberFormatsResponse;
import org.apache.fineract.client.models.PutAccountNumberFormatsRequest;
import org.apache.fineract.client.models.PutAccountNumberFormatsResponse;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface AccountNumberFormatApi {
  /**
   * Create an Account number format
   * Note: You may associate a single Account number format for a given account type Mandatory Fields for Account number formats accountType
   * @param postAccountNumberFormatsRequest  (optional)
   * @return Observable&lt;PostAccountNumberFormatsResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("accountnumberformats")
  Observable<PostAccountNumberFormatsResponse> create(
    @retrofit2.http.Body PostAccountNumberFormatsRequest postAccountNumberFormatsRequest
  );

  /**
   * Delete an Account number format
   * Note: Account numbers created while this format was active would remain unchanged.
   * @param accountNumberFormatId accountNumberFormatId (required)
   * @return Observable&lt;DeleteAccountNumberFormatsResponse&gt;
   */
  @DELETE("accountnumberformats/{accountNumberFormatId}")
  Observable<DeleteAccountNumberFormatsResponse> delete(
    @retrofit2.http.Path("accountNumberFormatId") Long accountNumberFormatId
  );

  /**
   * List Account number formats
   * Example Requests:  accountnumberformats   accountnumberformats?fields&#x3D;accountType,prefixType
   * @return Observable&lt;List&lt;GetAccountNumberFormatsIdResponse&gt;&gt;
   */
  @GET("accountnumberformats")
  Observable<List<GetAccountNumberFormatsIdResponse>> retrieveAll3();
    

  /**
   * Retrieve an Account number format
   * Example Requests:  accountnumberformats/1   accountnumberformats/1?template&#x3D;true   accountnumberformats/1?fields&#x3D;accountType,prefixType
   * @param accountNumberFormatId accountNumberFormatId (required)
   * @return Observable&lt;GetAccountNumberFormatsIdResponse&gt;
   */
  @GET("accountnumberformats/{accountNumberFormatId}")
  Observable<GetAccountNumberFormatsIdResponse> retrieveOne(
    @retrofit2.http.Path("accountNumberFormatId") Long accountNumberFormatId
  );

  /**
   * Retrieve Account number format Template
   * This is a convenience resource. It can be useful when building maintenance user interface screens for client applications. The template data returned consists of any or all of:  Field Defaults Allowed Value Lists  Example Request:  accountnumberformats/template
   * @return Observable&lt;GetAccountNumberFormatsResponseTemplate&gt;
   */
  @GET("accountnumberformats/template")
  Observable<GetAccountNumberFormatsResponseTemplate> retrieveTemplate2();
    

  /**
   * Update an Account number format
   * 
   * @param accountNumberFormatId accountNumberFormatId (required)
   * @param putAccountNumberFormatsRequest  (required)
   * @return Observable&lt;PutAccountNumberFormatsResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("accountnumberformats/{accountNumberFormatId}")
  Observable<PutAccountNumberFormatsResponse> update1(
    @retrofit2.http.Path("accountNumberFormatId") Long accountNumberFormatId, @retrofit2.http.Body PutAccountNumberFormatsRequest putAccountNumberFormatsRequest
  );

}

File: src/main/kotlin/org/openapitools/client/apis/AccountNumberFormatApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.DeleteAccountNumberFormatsResponse
import org.openapitools.client.models.GetAccountNumberFormatsIdResponse
import org.openapitools.client.models.GetAccountNumberFormatsResponseTemplate
import org.openapitools.client.models.PostAccountNumberFormatsRequest
import org.openapitools.client.models.PostAccountNumberFormatsResponse
import org.openapitools.client.models.PutAccountNumberFormatsRequest
import org.openapitools.client.models.PutAccountNumberFormatsResponse
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface AccountNumberFormatApi {
    /**
     * Create an Account number format
     * Note: You may associate a single Account number format for a given account type Mandatory Fields for Account number formats accountType
     * Responses:
     *  - 200: OK
     *
     * @param postAccountNumberFormatsRequest  (optional)
     * @return [PostAccountNumberFormatsResponse]
     */
    @POST("v1/accountnumberformats")
    suspend fun create(@Body postAccountNumberFormatsRequest: PostAccountNumberFormatsRequest? = null): PostAccountNumberFormatsResponse

    /**
     * Delete an Account number format
     * Note: Account numbers created while this format was active would remain unchanged.
     * Responses:
     *  - 200: OK
     *
     * @param accountNumberFormatId accountNumberFormatId
     * @return [DeleteAccountNumberFormatsResponse]
     */
    @DELETE("v1/accountnumberformats/{accountNumberFormatId}")
    suspend fun delete(@Path("accountNumberFormatId") accountNumberFormatId: Long): DeleteAccountNumberFormatsResponse

    /**
     * List Account number formats
     * Example Requests:  accountnumberformats   accountnumberformats?fields&#x3D;accountType,prefixType
     * Responses:
     *  - 200: OK
     *
     * @return [kotlin.collections.List<GetAccountNumberFormatsIdResponse>]
     */
    @GET("v1/accountnumberformats")
    suspend fun retrieveAll3(): List<GetAccountNumberFormatsIdResponse>

    /**
     * Retrieve an Account number format
     * Example Requests:  accountnumberformats/1   accountnumberformats/1?template&#x3D;true   accountnumberformats/1?fields&#x3D;accountType,prefixType
     * Responses:
     *  - 200: OK
     *
     * @param accountNumberFormatId accountNumberFormatId
     * @return [GetAccountNumberFormatsIdResponse]
     */
    @GET("v1/accountnumberformats/{accountNumberFormatId}")
    suspend fun retrieveOne(@Path("accountNumberFormatId") accountNumberFormatId: Long): GetAccountNumberFormatsIdResponse

    /**
     * Retrieve Account number format Template
     * This is a convenience resource. It can be useful when building maintenance user interface screens for client applications. The template data returned consists of any or all of:  Field Defaults Allowed Value Lists  Example Request:  accountnumberformats/template
     * Responses:
     *  - 200: OK
     *
     * @return [GetAccountNumberFormatsResponseTemplate]
     */
    @GET("v1/accountnumberformats/template")
    suspend fun retrieveTemplate2(): GetAccountNumberFormatsResponseTemplate

    /**
     * Update an Account number format
     *
     * Responses:
     *  - 200: OK
     *
     * @param accountNumberFormatId accountNumberFormatId
     * @param putAccountNumberFormatsRequest
     * @return [PutAccountNumberFormatsResponse]
     */
    @PUT("v1/accountnumberformats/{accountNumberFormatId}")
    suspend fun update1(
        @Path("accountNumberFormatId") accountNumberFormatId: Long,
        @Body putAccountNumberFormatsRequest: PutAccountNumberFormatsRequest
    ): PutAccountNumberFormatsResponse

}
